File: k8s/jaeger/query_api/python/main.py
import grpc
import sys

sys.path.append("/Users/hirokihanada/code/src/github.com/hanapedia/k8s_provisioner/k8s/jaeger/query_api_python/proto-gen-python")
import query_pb2_grpc as pb



def main():
    channel = grpc.insecure_channel("localhost:16686")
    stub = pb.QueryServiceStub(channel)
    # stub.FindTraces({
    #                     "service_name": "gateway",
    #                     "operation_name": "all",
    #                     "tags": {},
    #                     "start_time_min": 1670300574256000,
    #                     "start_time_max": 1670300574257000,
    #                     "duration_min": '',
    #                     "duration_max": '',
    #                     "search_depth": '',
    #                 })
    
    stub.GetServices({})
    # stub.GetTrace(request={"trace_id":bytes("d4741835c98b39a12a9bccf6a70e51ea", "utf-8")})

if __name__ == "__main__":
    main()

File: k8s/jaeger/query_api/go/main.go
package main

import (
	"context"
	"io"
	"log"
	"time"

	"google.golang.org/grpc"
	"google.golang.org/grpc/credentials/insecure"
	"github.com/jaegertracing/jaeger/model"

	pb "github.com/hanapedia/k8s_provisioner/k8s/jaeger/query_api/go/proto-gen-go"
)

func main(){
	var optsClient []grpc.DialOption
  optsClient = append(optsClient, grpc.WithTransportCredentials(insecure.NewCredentials()))

  conn, err := grpc.DialContext(context.Background(),"localhost:16685", optsClient...)
	if err != nil {
		log.Fatalf("Cannot establish connection with the server: %v", err)
	}
	defer conn.Close()

  queryServiceClient := pb.NewQueryServiceClient(conn)

  getTrace(queryServiceClient, "43f34352394cc9dc89404f4e31bdc6c6")
// log.Printf("min: %v", time.Now().Add(-3000000000))
  // log.Printf("min: %v", time.Now())
  // log.Printf("min: %v", time.UTC)
  // res, err := queryServiceClient.GetServices(context.Background(), &pb.GetServicesRequest{})
}


func getTrace(client pb.QueryServiceClient, traceIdstr string) {
  traceId, err := model.TraceIDFromString(traceIdstr)
  if err != nil {
    log.Fatalln("failed to parse trace id")
  }
  traceRequestParam := pb.GetTraceRequest{
    TraceID: traceId,
  }
  log.Println(traceRequestParam)
  stream, err := client.GetTrace(context.Background(), &pb.GetTraceRequest{})
	if err != nil {
    log.Fatalf("query failed: %v", err)
	}
  for {
      feature, err := stream.Recv()
      if err == io.EOF {
          break
      }
      if err != nil {
          log.Fatalf("%v.ListFeatures(_) = _, %v", client, err)
      }
      log.Println(feature)
  }
}

func findTraces(client pb.QueryServiceClient) {
  traceQueryParameters := pb.TraceQueryParameters{
    ServiceName: "gateway",
    OperationName: "all",
    StartTimeMin: time.Now().Add(-300000000000),
    StartTimeMax: time.Now(),
  }
  stream, err := client.FindTraces(context.Background(), &pb.FindTracesRequest{Query: &traceQueryParameters})
	if err != nil {
    log.Fatalf("query failed: %v", err)
	}
  for {
      feature, err := stream.Recv()
      if err == io.EOF {
          break
      }
      if err != nil {
          log.Fatalf("%v.ListFeatures(_) = _, %v", client, err)
      }
      log.Println(feature)
  }
}
